<<<<<<<< HEAD:package/python3/0004-Serial-ioctl-workaround.patch
From 52a85666f956ee11d5d165447dcc108e9034d78c Mon Sep 17 00:00:00 2001
========
From 0061a02bbe82c9b3b631d3b820bab0ba18482ebc Mon Sep 17 00:00:00 2001
>>>>>>>> upstream/master:package/python3/0007-Serial-ioctl-workaround.patch
From: Baruch Siach <baruch@tkos.co.il>
Date: Wed, 23 Dec 2015 11:44:02 +0100
Subject: [PATCH] Serial ioctl() workaround

The ioctls.h of some architectures (notably xtensa) references structs from
linux/serial.h. Make sure to include this header as well.

Also, undef TIOCTTYGSTRUCT that require reference to internal kernel tty_struct,
but isn't actually referenced in modern kernels.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
 Modules/termios.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Modules/termios.c b/Modules/termios.c
<<<<<<<< HEAD:package/python3/0004-Serial-ioctl-workaround.patch
index 402e6ac908a..688afd78994 100644
========
index 23771b2ce99..c1579af435d 100644
>>>>>>>> upstream/master:package/python3/0007-Serial-ioctl-workaround.patch
--- a/Modules/termios.c
+++ b/Modules/termios.c
@@ -15,7 +15,9 @@
 #endif
 
 #include <termios.h>
+#include <linux/serial.h>
 #include <sys/ioctl.h>
+#undef TIOCTTYGSTRUCT
 
 /* HP-UX requires that this be included to pick up MDCD, MCTS, MDSR,
  * MDTR, MRI, and MRTS (apparently used internally by some things
-- 
<<<<<<<< HEAD:package/python3/0004-Serial-ioctl-workaround.patch
2.43.0
========
2.44.0
>>>>>>>> upstream/master:package/python3/0007-Serial-ioctl-workaround.patch

